bitcoin/src/bip32.rs

```diff
@@ -730,62 +1088,73 @@ impl Xpub {
     /// the internal public key representation.
     pub fn to_x_only_pub(self) -> XOnlyPublicKey { XOnlyPublicKey::from(self.public_key) }
 
-    /// Attempts to derive an extended public key from a path.
-    ///
-    /// The `path` argument can be any type implementing `AsRef<ChildNumber>`, such as `DerivationPath`, for instance.
-    pub fn derive_pub<C: secp256k1::Verification, P: AsRef<[ChildNumber]>>(
+    /// Derive an extended public key from a path that contains NormalChildNumbers.
+    pub fn derive_pub<
+        'a,
+        C: secp256k1::Verification,
+        T: IntoIterator<Item = &'a NormalChildNumber>,
```

Comment:
This should also support non-reference iterators by requiring `Item: Borrow<NormalChildNumber>` instead of forcing a reference.